src/DataTransferObject.php

```diff
@@ -25,9 +25,14 @@ public function __construct(...$args)
         $class = new DataTransferObjectClass($this);
 
         foreach ($class->getProperties() as $property) {
-            $property->setValue($args[$property->name] ?? $this->{$property->name} ?? null);
-
-            unset($args[$property->name]);
+            if (array_key_exists($property->name, $args)) {
+                $property->setValue($args[$property->name]);
+                unset($args[$property->name]);
+            } elseif ($property->hasDefaultValue()) {
+                $property->setValue($property->getDefaultValue());
```

Comment:
If the property has a default value already, why should we re-set it?

Reply:
We don't necessarily need to set it here, but I inferred from previous behaviour that this was intended behaviour, per:

```
$property->setValue(... $this->{$property->name} ...);
```

I imagine without this line there would be a change in behaviour if the class is inherited and `setValue` is overloaded.